### PR TITLE
Use 1es templates

### PR DESCRIPTION
**Pull request by hallipr**

### CODE REVIEW

File: eng/pipelines/pr.yml [hunk 1]
```diff
@@ -4,57 +4,73 @@ pr:
...
-  - job: docker_build
-    pool:
-      name: azsdk-pool-mms-ubuntu-2004-general
-      vmImage: ubuntu-20.04
-    steps:
-      - script: docker build -f ./docker/Dockerfile .
-        displayName: Docker build
+            strategy:
+              matrix:
+                "Node 18.x":
```
Comment (timotheeguerin) [inline eng/pipelines/pr.yml:24]: This had to be split in 2 separate matrix?

**Review by timotheeguerin — APPROVED** [review]

File: eng/pipelines/pr.yml (hunk 1 above)
Comment (hallipr) [reply eng/pipelines/pr.yml:24]: The `os:` parameter in `pool` has to be a compile time value, so we can't use a matrix value there

**Pull request merged**